src/components/meal/FavoriteMealModal.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+import {
+   Button,
+   Card,
+   DefaultTheme,
+   Modal,
+   Paragraph,
+   Portal,
+   Title,
+} from 'react-native-paper';
+import * as Yup from 'yup';
+import { Formik, FormikValues } from 'formik';
+
+// Custom components
+import Checkbox from '../Checkbox';
+import TextInput from '../TextInput';
+
+interface FavoriteMealModalProps {
+   isVisible: boolean;
+   onDismiss: () => void;
+   onSetMealName: (mealName: string) => void;
+}
+
+const FavoriteMealModal = (props: FavoriteMealModalProps): JSX.Element => {
+   const [inputMealName, setInputMealName] = useState('');
+   const [isSavingFavoriteMeal, setIsSavingFavoriteMeal] = useState(false);
+
+   //    const MealNameSchema = Yup.object().shape({
```

Comment:
Note for future us: Leaving this commented since we are aiming for basically functionality for the end of semester **demo**